x-pack/plugins/monitoring/server/alerts/missing_data_alert.ts

```diff
@@ -0,0 +1,526 @@
+      description: i18n.translate(
+        'xpack.monitoring.alerts.missingData.actionVariables.internalShortMessage',
+        {
+          defaultMessage: 'The short internal message generated by Elastic.',
+        }
+      ),
+    },
+    {
+      name: 'internalFullMessage',
+      description: i18n.translate(
+        'xpack.monitoring.alerts.missingData.actionVariables.internalFullMessage',
+        {
+          defaultMessage: 'The full internal message generated by Elastic.',
+        }
+      ),
+    },
+    {
+      name: 'state',
+      description: i18n.translate('xpack.monitoring.alerts.missingData.actionVariables.state', {
+        defaultMessage: 'The current state of the alert.',
+      }),
+    },
+    {
+      name: 'stackProducts',
+      description: i18n.translate(
+        'xpack.monitoring.alerts.missingData.actionVariables.stackProducts',
+        {
+          defaultMessage: 'The stack products missing data.',
+        }
+      ),
+    },
+    {
+      name: 'count',
+      description: i18n.translate('xpack.monitoring.alerts.missingData.actionVariables.count', {
+        defaultMessage: 'The number of stack products missing data.',
+      }),
+    },
+    {
+      name: 'clusterName',
+      description: i18n.translate(
+        'xpack.monitoring.alerts.missingData.actionVariables.clusterName',
+        {
+          defaultMessage: 'The cluster to which the stack products belong.',
+        }
+      ),
+    },
+    {
+      name: 'action',
+      description: i18n.translate('xpack.monitoring.alerts.missingData.actionVariables.action', {
+        defaultMessage: 'The recommended action for this alert.',
+      }),
+    },
+    {
+      name: 'actionPlain',
+      description: i18n.translate(
+        'xpack.monitoring.alerts.missingData.actionVariables.actionPlain',
+        {
+          defaultMessage: 'The recommended action for this alert, without any markdown.',
+        }
+      ),
+    },
+  ];
+
+  protected async fetchData(
+    params: CommonAlertParams,
+    callCluster: any,
+    clusters: AlertCluster[],
+    uiSettings: IUiSettingsClient,
+    availableCcs: string[]
+  ): Promise<AlertData[]> {
+    let indexPattern = appendMetricbeatIndex(this.config, INDEX_PATTERN);
+    if (availableCcs) {
+      indexPattern = getCcsIndexPattern(indexPattern, availableCcs);
+    }
+    const duration = parseDuration(((params as unknown) as MissingDataParams).duration);
+    const limit = parseDuration(((params as unknown) as MissingDataParams).limit);
+    const missingData = await fetchMissingData(
+      callCluster,
+      clusters,
+      indexPattern,
+      limit,
+      this.config.ui.max_bucket_size
+    );
+    return missingData.map((missing) => {
+      return {
+        instanceKey: `${missing.clusterUuid}:${missing.stackProduct}:${missing.stackProductUuid}`,
+        clusterUuid: missing.clusterUuid,
+        shouldFire: missing.gapDuration > duration && missing.gapDuration <= limit,
```

Comment:
I don't think you need: `... && missing.gapDuration <= limit` check, since your search query is already within the limit range (give or take). But, even if, wouldn't it still qualify as a valid trigger? Since, `duration` would always be less than `limit`